File: java/pe/src/main/java/ktn/pe/Pe3.java
package ktn.pe;

import ktn.pe.util.PeUtils;

import java.util.ArrayList;
import java.util.List;

public class Pe3 implements Pe {
    private long n;

    public Pe3() {
        super();

        n = 600851475143L;
    }

    @Override
    public void setArgs(String[] args) {
        if (args.length > 0) {
            n = Long.parseLong(args[0]);
        }
    }

    @Override
    public void solve() {
        List<Long> divisors = new ArrayList<>();
        PeUtils.divisors(divisors, n);
        System.out.println(PeUtils.format(this, divisors));
    }

    @Override
    public void run() {
        solve();
    }

    @Override
    public int getProblemNumber() {
        return 3;
    }
}

File: kt/pe/src/main/kotlin/ktn/pe/Pe3.kt
package ktn.pe

class Pe3 : Pe {
    private var n: Long = 0

    override val problemNumber: Int
        get() = 3

    init {
        n = 600851475143L
    }

    override fun setArgs(args: Array<String>) {
        if (args.isNotEmpty()) {
            n = args[0].toLong()
        }
    }

    override fun solve() {
        val divisors = mutableListOf<Long>()
        PeUtils.divisors(divisors, n)
        System.out.println(PeUtils.format(this, divisors))
    }

    override fun run() {
        solve()
    }
}
